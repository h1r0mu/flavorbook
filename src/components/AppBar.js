import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { createMuiTheme } from "@material-ui/core/styles";
import EmojiFoodBeverageIcon from "@material-ui/icons/EmojiFoodBeverage";
import LaptopMacIcon from "@material-ui/icons/LaptopMac";
import ShareIcon from "@material-ui/icons/Share";
import ChatIcon from "@material-ui/icons/Chat";
import StoreIcon from "@material-ui/icons/Store";


const styles = {
  root: {
    flexGrow: 1,
    backgroundColor: "transparent",
  },
  menuButton: {
    marginRight: 20,
    marginLeft: -12,
  },
  title: {
    flexGrow: 1,
  },
  drawerHeader: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
  },
  drawerInner: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
  },
  coffee: {
    marginTop: 0,
    marginLeft: 20,
    fontSize: 20,
  },
};

class ButtonAppBar extends React.Component {
  constructor() {
    super();
    this.handleDrawerOpen = this.handleDrawerOpen.bind(this);
    this.handleDrawerClose = this.handleDrawerClose.bind(this);
    this.state = { drawerIsOpen: false };
  }

  handleDrawerOpen() {
    this.setState({ drawerIsOpen: true });
  }

  handleDrawerClose() {
    this.setState({ drawerIsOpen: false });
  }

  render() {
    return (
        <div className={this.props.classes.root}>
          <AppBar position="static">
            <Toolbar>
              <IconButton
                edge="start"
                className={this.props.classes.menuButton}
                color="inherit"
                aria-label="menu"
                onClick={this.handleDrawerOpen}
              >
                <MenuIcon />
              </IconButton>
              <Typography variant="h6" className={this.props.classes.title}>
                Coffee Flavors
                <EmojiFoodBeverageIcon className={this.props.classes.coffee} />
              </Typography>
              <Button color="inherit">Login(coming soon)</Button>
            </Toolbar>
          </AppBar>
          <Drawer
            variant="temporary"
            classes={{
              paper: this.props.classes.drawerPaper,
            }}
            onEscapeKeyDown={this.handleDrawerClose}
            onBackdropClick={this.handleDrawerClose}
            open={this.state.drawerIsOpen}
          >
            <div className={this.props.classes.drawerHeader}>
              <IconButton onClick={this.handleDrawerClose}>
                <MenuIcon />
              </IconButton>
            </div>
            <div className={this.props.classes.drawerInner}>
              <List>
                {[
                  "豆の香りを調べる",
                  "メンバーの香りをみてみる(近日公開)",
                  "メンバー同士で話しあう(近日公開)",
                  "豆のお店を探す(近日公開)",
                ].map((text, index) => (
                  <ListItem button key={text}>
                    <ListItemIcon>
                      {index % 4 === 0 ? (
                        <LaptopMacIcon />
                      ) : index % 4 === 1 ? (
                        <ShareIcon />
                      ) : index % 4 === 2 ? (
                        <ChatIcon />
                      ) : (
                        <StoreIcon />
                      )}
                    </ListItemIcon>
                    <ListItemText primary={text} />
                  </ListItem>
                ))}
              </List>
            </div>
          </Drawer>
        </div>
    );
  }
}

ButtonAppBar.propTypes = {
  drawerIsOpen: PropTypes.bool,
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(ButtonAppBar);
