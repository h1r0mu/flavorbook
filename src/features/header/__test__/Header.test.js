import AppBar from "../Header";
import React from "react";
import renderer from "react-test-renderer";
import {createMemoryHistory} from 'history'
import {BrowserRouter } from "react-router-dom";

//const history = createMemoryHistory();
//history.push("/login")
it("renders correctly", () => {
  const appBar = renderer.create(
				<BrowserRouter>
				 <AppBar />
				</BrowserRouter>
		).toJSON();
  expect(appBar).toMatchSnapshot();
});
