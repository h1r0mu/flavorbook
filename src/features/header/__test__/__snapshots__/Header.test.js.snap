// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="menu"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        Coffee Flavors
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root makeStyles-coffee-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 3H9v2.4l1.81 1.45c.12.09.19.24.19.39v4.26c0 .28-.22.5-.5.5h-4c-.28 0-.5-.22-.5-.5V7.24c0-.15.07-.3.19-.39L8 5.4V3H4v10c0 2.21 1.79 4 4 4h6c2.21 0 4-1.79 4-4v-3h2c1.11 0 2-.9 2-2V5c0-1.11-.89-2-2-2zm0 5h-2V5h2v3zM4 19h16v2H4z"
          />
        </svg>
      </h6>
      <a
        className="makeStyles-link-7"
        href="/login"
        onClick={[Function]}
      >
        <button
          aria-haspopup="true"
          aria-label="account of current user"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
            </svg>
          </span>
        </button>
      </a>
    </div>
  </header>
</div>
`;
